networkx/classes/digraph.py

```diff
@@ -970,7 +970,34 @@ def in_edges(self):
         in_edges : InEdgeView
             A view of edge attributes, usually it iterates over (u, v)
             or (u, v, d) tuples of edges, but can also be used for
-            attribute lookup as `edges[u, v]['foo']`.
+            attribute lookup as `edges[u, v]['foo']`. The *EdgeView family
+            of objects handle requests for all edges without any data. The
+            *EdgeDataView family of objects handle special requests: either
+            restricting to source nodes via nbunch or asking for edge data
+            via data. Also note the difference between G.edges and G.edges()
+            in the examples. G.edges reports edge tuples. If they are multiedges,
+            then the edge tuples are 3-tuples: (n, nbr, edgekey). But if they are
+            not multiedges they are 2-tuples: (n, nbr). But G.edges() returns 2-tuples
+            whether they are multiedges or not.
```

Comment:
Thanks for the proposed wording here - I agree that this is correct, but is probably too much information for this section. IMO this would be a better fit in the docstring for the EdgeView classes (though finding which one(s) exactly will be a challenge).

Instead of adding all of this information here, I'd propose to simply update the return types to include `InEdgeDataView` as well. In other words, update the line a few above to `in_edges : InEdgeView or InEdgeDataView`.

```suggestion
            attribute lookup as ``edges[u, v]['foo']``.
```